### DISCUSSION TITLE
CloudFlare WARP + Marzban

### DISCUSSION THREAD

**Discussion by 0xAsghar** [Gozargah/Marzban#141] [Ideas]
سلام، با عرض خسته نباشید خدمت تیم مرزبان
آموزش استفاده از کلودفلر Warp جهت رفع مشکلات مربوط به آیپی سرورها مخصوصا آیپی های دیتاسنتر هتزنر:

### مرحله اول: نصب و کانفیگ مرزبان
شما از دو [روش دستی ](https://github.com/Gozargah/Marzban#manual-install-advanced)و [از طریق داکر](https://github.com/Gozargah/Marzban-examples) می توانید مرزبان را نصب کنید.
پیشنهاد من به شما استفاده از [نسخه fully-single-port](https://github.com/Gozargah/Marzban-examples/blob/master/fully-single-port) می باشد.

### مرحله دوم: تنظیمات مربوط به xray و env
**تنظیمات xray**
`cd marzban`
`nano xray_config.json`

**تنظیمات env**
`cd marzban`
`nano env`

**اعمال تغییرات**
`cd marzban`
`docker compose down`
`docker compose up -d`

### مرحله سوم: نصب اسکریپت warp
1. با دستور زیر وارد root شوید:
`cd ~`
2. کد زیر را اجرا کنید تا نصب انجام شود:
 `bash <(curl -fsSL git.io/warp.sh) proxy`

### مرحله چهارم: تغییرات لازم در xray_config
وارد پوشه مرزبان شوید:
`cd marzban`
با ویرایشگر nano فایل xray_config را باز کنید:
`nano xray_config.json`

در قسمت outbounds کد زیر را اضافه کنید:
       {
    "tag": "socks5-warp",
    "protocol": "socks",
    "settings": {
        "servers": [{
            "address": "127.0.0.1",
            "port": 40000
                   }]
          }
       }

 **:نمونه** 
![outbounds](https://user-images.githubusercontent.com/30578387/227958899-4d6c20c1-08f8-4cf3-be05-827391d4fe71.png)

و در routing و قسمت rules کد زیر را اضافه کنید:
               {
                    "type": "field",
                    "outboundTag": "socks5-warp",
                    "domain": ["ipinfo.io","iplocation.net","google.com","spotify.com"]
                }
**نمونه:** 
![routing](https://user-images.githubusercontent.com/30578387/227959011-27686909-e69b-4da3-b9b0-2afa17de8f6c.png)

**توجه:** من در اینجا چهار دامنه قرار دادم شما میتوانید دامنه هایی که قصد دارید از warp استفاده بشه را اینجا تفییر یا اضافه کنید.

اگر سوالی داشتید خوشحال میشم کمک کنم.

**Comment by ItsElw** [comment]
ممنون،خیلی خوب توضیح دادید🌹

**Reply by MrHosseinS** [reply]
منابع رو پاره میکنه...

**Reply by heartbreath180** [reply]
Hello, there is a login section for help. I wish I could tell people that I forgot my password, but they ask me again for the password. By the way, do you think that if it is not free, we are willing to log in, but it is all for the account you are asking for, I wish you would have said that this is the first thing.  Let's not waste time. I can log in. My password is correct, but it won't accept. I'm sure it's correct.

**Reply by heartbreath180** [reply]
چرا ورود یا ایجاد حساب زو انقدر سخت کردین.من حتی نتونستم وارد بشم.ببینم حساب دارم یا نه.لطفا کمک کنید اما زود مرسی

**Reply by heartbreath180** [reply]
اگر رایگان نیست چرا اطلاع نمیدین .گیج شدم واقعا
غکر کنم العان در جواب یا نظر کسی دارم انتقاد میکنم حتی واصلن فکر نکنید که چون نتونستم با پشتیبانی تماس بیرم اینکارو کردم نه اصلن اینطور فکر نکنید باشه.عجبا

**Comment by nasiriscript** [comment]
این آموزش خیلی خوبه جاش خالی بود.  سوال، وارپ محدودیت حجمی داره درسته؟ چقدره محدودیتش؟ اگه محدودیت نداره به طریقی مثلا regex کل ترافیک رو انداخت روش؟
با تشکر.

**Reply by M03ED** [reply]
محدودیت حجمی نداره ولی ورژن رایگان سرعت از حد خاصی بالاتر نمیره و در ضمن استفاده ازش برای کل دیتا باعث مصرف زیاد منابع میشه و به سرور فشار زیادی میاد

**Reply by nasiriscript** [reply]
مچکرم، متاسفانه سرعت دانلود سرور رو 250 مگابایت قفل میشه اون مشکل مصرف منابع رو فکر کنم warp-go نداشته باشه ولی با این سرعت تعداد محدودی باید استفاده کنند.

**Comment by M03ED** [comment]
سلام
چطور باید warp رو برای کل داده ها استفاده کنیم ؟

**Reply by nasiriscript** [reply]
> سلام چطور باید warp رو برای کل داده ها استفاده کنیم ؟

از این استفاده کن
https://github.com/fscarmen/warp/blob/main/README.md#warp-go-%E8%BF%90%E8%A1%8C%E8%84%9A%E6%9C%AC
فقط dualstack من انتخاب کردم سرعت محدود میشد ولی رو ipv4 خالی ok بود

**Reply by alircxa** [reply]
@nasiriscript متاسفانه این لینک مسدود شده. جای دیگه‌ای نداشتیدش؟

**Reply by nasiriscript** [reply]
https://gitlab.com/fscarmen/warp

**Reply by heartbreath180** [reply]
برنامه مضخرفی هست مثل صاحبانش

**Comment by hosseynkebria** [comment]
سلام
چطور کاری کنم آی پی سرور خودمو برگردونه؟
همه سایت ها آیپیم شده ماله کلاودفلر

**Reply by M03ED** [reply]
احتمال Warp رو اولین مورد توی outbound ها قرار دادید ، ببریدش آخرین مورد حل میشه

**Reply by hosseynkebria** [reply]
mamnon hal shod ❤❤

**Comment by hosseynkebria** [comment]
رم خیلی استفاده می کنه ، روش بهینه تر وجود داره؟

**Comment by SaintShit** [comment]
سلام
این کار رو میتونید بهینه تر انجام بدید

ابتدا [wgcf](https://github.com/ViRb3/wgcf) رو نصب کنید

```
wget https://github.com/ViRb3/wgcf/releases/download/v2.2.16/wgcf_2.2.16_linux_amd64
chmod +x wgcf_2.2.16_linux_amd64
mv wgcf_2.2.16_linux_amd64 /usr/bin/wgcf
rm wgcf_2.2.16_linux_amd64
```
بعدش این دستور رو بزنید و حساب وارپ بسازید
```
wgcf register
wgcf generate
```
حالا یک فایل پروفایل وایرگارد ساخته براتون `wgcf-profile.yaml`
بازش کنید و یه جا توی بخش `[Interface]` آپشن

```
[Interface]
...
Table = off

[Peer]
...
```
رو اضافه کنید.

**حواستون باشه این مرحله اضافه کردن Table = off خیلی مهمه وگرنه بعد از استارت کردن سرویس کل سرورتون به وارپ وصل میشه و دیگه نمیتونین ssh بزنید و مجبور میشید از طریق سریال پورت از پنل کلاد سرویستون وصل بشین.**

حالا ابزار های مورد نیاز وایرگارد رو نصب کنید

```sudo apt install wireguard-dkms wireguard-tools resolvconf```

بعدش پروفایل رو کپی کنید توی `/etc/wireguard/`

```sudo cp wgcf-profile.conf /etc/wireguard/warp.conf```

و سرویس وایرگارد رو فعال کنید

```sudo systemctl enable --now wg-quick@warp```


حالا اگر دستور 

```ip a```

رو بزنید باید اینترفیس `warp` رو توی لیست ببینید

توی فایل xray_config.json اوت باندی که میخواین ترافیکش از وارپ رد بشه رو ادیت کنید و این گزینه هارو به `streamSettings` اش اضافه کنید
```json
"streamSettings": {
    "sockopt": {
        "tcpFastOpen": true,
        "interface": "warp"
    }
}
```
مثلا اوتباند `freedom` رو میتونید کاملا از وارپ رد کنید که یعنی همه ترافیک خروجی پروکسی میره روی وارپ

```json
{
    "tag": "direct",
    "protocol": "freedom",
    "streamSettings": {
        "sockopt": {
            "tcpFastOpen": true,
            "interface": "warp"
        }
    }
}
```

همین دیگه.

**Reply by AC-Lover** [reply]
![image](https://user-images.githubusercontent.com/49290111/235386562-c73f50ac-9968-41ff-a3aa-ac5f8cc2b5c6.png)

![image](https://user-images.githubusercontent.com/49290111/235386569-d3cbaeb4-947e-43fb-bca1-503f9a258b48.png)

![image](https://user-images.githubusercontent.com/49290111/235386574-96c48fcc-7258-499b-b4cd-fdd41060ebbe.png)


مشکل چیست که کار نمیکنه؟

**Reply by M03ED** [reply]
پروتوکل رو به جای socks5 روی freedom بزار 

![Screenshot_2023-05-01-07-16-11-137_com.server.auditor.ssh.client.jpg](https://user-images.githubusercontent.com/50927468/235403732-f98cc1bf-d704-43d6-883d-9c93488cbaa3.jpg)

**Reply by AC-Lover** [reply]
> پروتوکل رو به جای socks5 روی freedom بزار
> 
> ![Screenshot_2023-05-01-07-16-11-137_com.server.auditor.ssh.client.jpg](https://user-images.githubusercontent.com/50927468/235403732-f98cc1bf-d704-43d6-883d-9c93488cbaa3.jpg)

آها من قصدم این بود ک کل ترافیک رو رد نکنم ولی به نظر میرسه که بازم باید از 
```
{
    "tag": "direct",
    "protocol": "freedom",
    "streamSettings": {
        "sockopt": {
            "tcpFastOpen": true,
            "interface": "warp"
        }
    }
}
```

استفاده کنم
تست میکنم ممنون

**Reply by M03ED** [reply]
خیر لازم نیست ، بزارش پایین ترین و تگ رو هم عوض کن مشکل نداره

**Comment by vipnets** [comment]
سلام وقتی ترافیک گوگل از وارپ رد میکنم تبلیغات که قبلا بسته بودم تبلیغات گوگل منظورم باز میشه و نمایش داده میشه 

راهی هست که تبلیغات گوگل و یوتیوب بسته بشه وقتی گوگل را با وارپ برای من باز میکنه

**Reply by NimaTarlani** [reply]
دی ان اس adguard بزنی جواب میده فک کنم

**Comment by tbehzad76** [comment]
لطفا تظیمات وارپ رو برای نود هم بزارین

**Comment by mhyrdarzi** [comment]
سلام
چطور میشه حذفش کرد؟

**Comment by mohammadi-hamid** [comment]
چطور میشه روی نودها هم warp رو گذاشت؟ چون الان وقتی به سرور اصلی وصل میشم گوگل و ... اکی هست اما وقتی به نود وصل میشم نه.

**Comment by deaduchiha** [comment]
دمت گرم 🤍

**Comment by Muhammad-heydari** [comment]
سلام چطوری میتونم اون ادرس وارپ رو عوض کنم؟

**Comment by loopy-iri** [comment]
اگه بخوام وارپ برای یوزر خاصی فعال باشه باید چیکار کنم؟

**Comment by arminahari-dev** [comment]
این اسکریپت که گذاشتید آرشیو شده و موقع نصب ارور میده 
راه حل جایگزین چیه؟
